### ISSUE TITLE
Fix `declaration-property-value-no-unknown` for multiple `<declaration-value>` in `typesSyntax`

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#6636]
### What steps are needed to reproduce the bug?

```js
testRule({
	ruleName,
	config: [
		true,
		{
			propertiesSyntax: { top: '| <foo()>' },
			typesSyntax: { 'foo()': 'foo( <declaration-value> , <declaration-value> )' },
		},
	],

	accept: [
		{
			code: 'a { top: foo(10px, 10%); }',
		},
	],
});
```

```js
testRule({
	ruleName,
	config: [
		true,
		{
			propertiesSyntax: { top: '| <foo()>' },
			typesSyntax: { 'foo()': 'foo( <declaration-value> <declaration-value> )' },
		},
	],

	accept: [
		{
			code: 'a { top: foo(10px 10%); }',
		},
	],
});
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "declaration-property-value-no-unknown": true
}
```

### How did you run Stylelint?

CLI and VSCode

### Which version of Stylelint are you using?

15.0.0

### What did you expect to happen?

I expected to be able to add custom syntax definitions with the catch all `<declaration-value>`.

Or more abstract :
I expected to be able to define a syntax that allows any component that is valid in a declaration value.

### What actually happened?

Various errors but the custom provided syntax never matches the values.

I might have misunderstood `<declaration-value>`

**Comment by jeddy3** [comment]
@romainmenke Thanks for the report and for using the template.

To allow any number of arguments of any type, we can use:

```json
{
  "rules": {
    "declaration-property-value-no-unknown": [true, {
      "propertiesSyntax": { "top": "| <foo()>" },
      "typesSyntax": { "foo()": "foo(<declaration-value>)" }
    }]
  }
}
```

([See demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYKAuEAO8UDMIIAKARgAZsAPAGhQoFIBKAbhQF8QaQBnATwB2GJFQTgePLuAgD8ASwDmY4AB0BKFCpAAnAK4AbGDy15V6jZpCww+pNqQY5MgLTZtOGNox9nANyT6ujDOAhDOugIA1qEA7gImKADaGHowNGoqGJlmmblZIG4eXnJGAMqCwlQJ6FpY2AlaAD4oADyEJEwAfFoc6fFZA7V82GUVItWW7cRMDSBTLda29o4u-oEwnTMgHBkD7AC6ahrsapzsQA).)

And to allow a specific number of arguments of a specific type, we can use:

```json
{
  "rules": {
    "declaration-property-value-no-unknown": [true, {
      "propertiesSyntax": { "top": "| <foo()>" },
      "typesSyntax": { "foo()": "foo(<length-percentage>#{2})" }
    }]
  }
}
```

However, the rule flags, as you say, for a specific number of arguments of any type.

I've labelled as needs investigation, in case anyone has time to dig deeper.

**Comment by romainmenke** [comment]
Just realised that I can declare them multiple times and refer to the property :

```js
{
	propertiesSyntax: {
		color: '| if( <custom-property-name> <\'color\'> else <\'color\'> )',
		'background-color': '|if( <custom-property-name> <\'background-color\'> else <\'background-color\'> )',
	},
	typesSyntax: {
		'if( <custom-property-name> <\'color\'> else <\'color\'> )': 'if( <custom-property-name> <\'color\'> else <\'color\'> )',
		'if( <custom-property-name> <\'background-color\'> else <\'background-color\'> )': 'if( <custom-property-name> <\'background-color\'> else <\'background-color\'> )',
	},
},
```

This way each property has correct syntax checking when using this custom function.
We only use it on things like `color`, so the repetition is fine :)

**Comment by jeddy3** [comment]
> This way each property has correct syntax checking when using this custom function.

👍 

Shall we close the issue?

**Comment by romainmenke** [comment]
Yes, good to clone :)
